### PR TITLE
Pal 232 jvm performance tests

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: services-manager/src/main/resources/application-warn.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/application-warn.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: services-manager/src/main/resources/application-examplelibs.yaml [hunk 2]
```diff
@@ -17,23 +17,22 @@ spring:
...
+      - examplelibs-config
 
 # Spring profile dependency resolution forces this two-profile setup
 # `example` profile must forcibly override its dependencies as spring.profiles.include properties can no longer be overridden
 # So, include `example-config` along with dependencies and force correct override order to happen there
 ---
 
 spring:
-  profiles: example-config
+  profiles: examplelibs-config
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/application-examplelibs.yaml]: naming convention should be example-libs-config

File: services-manager/src/main/resources/application-examplemodel.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+spring:
+  profiles:
+    include:
+      - examplelibs
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/application-examplemodel.yaml]: naming convention should be example-libs and example-model

File: services-manager/src/main/resources/application-exampleperf.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+spring:
+  profiles:
+    include:
+      - warn
+      - examplemodel
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/application-exampleperf.yaml]: ditto

File: services-manager/src/main/resources/application-exampleperf.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      - exampleperf-config
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/application-exampleperf.yaml]: example-perf-config

File: services-manager/src/main/resources/doc/README-TEMPLATE.md [hunk 6]
```diff
@@ -37,47 +37,117 @@ The Palisade Services Manager is a configuration-driven process-spawner and REST
...
+java -jar -Dspring.profiles.active=examplemodel executable.jar
+```
+ * Ensure first to start the discovery-service as above
+ * Services will start up with their cache/persistence-store prepopulated with example data
+ * The rest-example will run once all services have started
+ * Check `rest-example.log` for output data
+
+The data used in this example comes checked-in to the repo and does not need generating
+ 
+#### `exampleperf` - Automated execution of Palisade performance tests on example data (see [performance](https://github.com/gchq/Palisade-examples/tree/develop/performance))
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/doc/README-TEMPLATE.md]: example-perf

File: services-manager/src/main/resources/doc/README-TEMPLATE.md [hunk 7]
```diff
@@ -37,47 +37,117 @@ The Palisade Services Manager is a configuration-driven process-spawner and REST
...
+**The choice here between `eureka` or `static` profiles will be referred to unilaterally as the `environment` profile - make sure to substitute as appropriate.**  
+
+#### `examplelibs` - Pre-populated Palisade example (see [example-library](https://github.com/gchq/Palisade-examples/tree/develop/example-library))
+```bash
+java -jar -Dspring.profiles.active=examplelibs executable.jar
+```
+ * Ensure first to start the discovery-service as above
+ * Services will start up with their cache/persistence-store prepopulated with example data
+
+#### `examplemodel` - Automated execution of Palisade client on example data (see [example-model](https://github.com/gchq/Palisade-examples/tree/develop/example-model))
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/doc/README-TEMPLATE.md]: example-model

File: services-manager/src/main/resources/doc/README-TEMPLATE.md [hunk 8]
```diff
@@ -37,47 +37,117 @@ The Palisade Services Manager is a configuration-driven process-spawner and REST
...
+java -jar -Dspring.profiles.active=discovery executable.jar # start discovery-service without eureka (it is eureka)
+java -jar -Dspring.profiles.active=eureka executable.jar    # start all other services with eureka
+```
+ * Once the manager has finished running, check the eureka dashboard at `localhost:8083`
+ * These two commands will only exit once all services are ready, so the two `java -jar ...` commands can be chained together as `java .. discovery -jar .. && java .. eureka -jar ...`
+
+***All further profiles require the discovery service to be running, or require some tweaking to use the static setup.***  
+**The choice here between `eureka` or `static` profiles will be referred to unilaterally as the `environment` profile - make sure to substitute as appropriate.**  
+
+#### `examplelibs` - Pre-populated Palisade example (see [example-library](https://github.com/gchq/Palisade-examples/tree/develop/example-library))
```
Comment (developer01189998819991197253) [inline services-manager/src/main/resources/doc/README-TEMPLATE.md]: example-libs

**Review by developer01189998819991197253 — COMMENTED** [review]
naming convention change required

File: services-manager/src/main/resources/application-examplelibs.yaml (hunk 2 above)
Comment (dev-958) [reply services-manager/src/main/resources/application-examplelibs.yaml]: I would prefer example-config. It describes a state.

Comment (pd104923) [reply services-manager/src/main/resources/application-examplelibs.yaml]: I'd agree with @dev-958 that this should be left as example-config

File: services-manager/src/main/resources/application-example-model.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+spring:
+  profiles:
+    include:
+      - example-libs
+      - example-model-config
```
Comment (pd104923) [inline services-manager/src/main/resources/application-example-model.yaml:20]: While example-model is the module I'm not sure that this is the best name for the profile. Can we make it a more descriptive name? Especially as the module name might be changed in the future.

Comment (dev930018) [reply services-manager/src/main/resources/application-example-model.yaml:20]: Pending rename of the module.
For now it makes most sense that it matches the module name, but should we rename to a more-appropriate example-service or similar I could see this changing to match.

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by ac74475 — APPROVED** [review]

**Review by pd104923 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**